### ISSUE TITLE
Switcher issue

### ISSUE THREAD

**Issue by x-oscarr** [open-admin-org/open-admin#48]
**Switcher ignore value update**
Since the last update changed method `\OpenAdmin\Admin\Form\Field\SwitchField::prepare()`, new conditions in the method expect to take 'on' or 'off' strings (or int: 0 ,1), but get a bool value.

**To Reproduce**
1. Create the switcher
![image](https://user-images.githubusercontent.com/50730757/217671392-2a10bd30-9cb5-4ce0-994c-a326602b7896.png)

2. Try to switch the parameter 

Demonstration: 

https://user-images.githubusercontent.com/50730757/217674905-e3839225-7387-43d2-ba17-5f679c26dbae.mp4

**My investigation**
1. resources/views/grid/inline-edit/switch-script.blade.php
Script sends a request with the bool value that gives from the checked property of an input element. 
Property **checked** is always bool (as I know)
![image](https://user-images.githubusercontent.com/50730757/217671997-b2aac318-4b34-4662-88b7-cf71b90d20d7.png)

2. request
![image](https://user-images.githubusercontent.com/50730757/217672339-5732883d-0a7f-42ec-add4-fcec589351d7.png)

3. This value takes by the `prepare()` method of the field **SwitchField**
but after updates conditions have been changed, and now value false cannot be passed...
![image](https://user-images.githubusercontent.com/50730757/217672904-15b8507f-be1d-47a1-a8fb-627f975ba90e.png)
[Here is old method `prepare()` ](https://github.com/open-admin-org/open-admin/commit/dffebb5efb924796c479d75c294a55532b66579a#diff-648d7cacd0a5efde0e72d2c49292d722180e686e35b2e98efab779c0e44ca008R11)

4. ...cannot be passed because value with false will be ignored in `src/Form.php:863`
![image](https://user-images.githubusercontent.com/50730757/217673522-ab5ad7d3-62d6-4da9-b405-55e59dcd68fc.png)

**System**
- Open-admin version: 1.0.26
- PHP version: 8.1.14
- Laravel Version: 9.48.0
- OS: Win + WLS2 (Linux) - local server launched on WSL2
- Browser: Chrome

**Comment by x-oscarr** [comment]
I`m not sure why this method has been changes, but if the conditions should have the same structure, would be better to add a condition to resources/views/grid/inline-edit/switch-script.blade.php like a 
`var value = this.checked ? 'on' : 'off';` on smth like that

**Comment by RCyrix** [comment]
Hi there 

Please see pull request 

Tested and working on both Grid (Switch displayer) and Form

**Comment by open-admin-org** [comment]
Hi All, 
Sorry for late response. Thanks for the detailed report @x-oscarr and @RCyrix thanks for the merge request. I did some test and unfortunate it was not fully working in all cases. I ended up with the the fix that x-oscarr supplied. with the "on" : "off" instead of true.

Case:
1) Inline edits in the $grid depend on the $form.
2) If a field has a prepare function it will always be prepared before saving the model to the database. 
3) In case a list has one or more inline edit fields, other fields with a prepared function will be prepared.
4) If the $value !== false it will be ignore
5) if we put the value of the switch to 0 when its false (not supplied), it will also default to 0 when other fields are save.
